File: src/com/tjoris/timekeeper/ConfirmationDialog.java
package com.tjoris.timekeeper;

import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.DialogInterface;
import android.os.Bundle;

public abstract class ConfirmationDialog extends DialogFragment
{
	private int fMessage;
	private int fPositiveMessage;
	private int fNegativeMessage;
	
	public ConfirmationDialog()
	{
	}
	
	public void setOptions(final int message, final int positiveMessage, final int negativeMessage)
	{
		fMessage = message;
		fPositiveMessage = positiveMessage;
		fNegativeMessage = negativeMessage;
	}
	
	@Override
	public Dialog onCreateDialog(final Bundle savedInstanceState)
	{
		final AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
		builder.setMessage(fMessage).setPositiveButton(fPositiveMessage, new DialogInterface.OnClickListener()
		{
			public void onClick(final DialogInterface dialog, final int id)
			{
				confirm();
			}
		}).setNegativeButton(fNegativeMessage, null);
		return builder.create();
	}
	
	public abstract void confirm();
}

File: TimeKeeperAndroid/app/src/main/java/be/t_ars/timekeeper/ConfirmationDialog.kt
package be.t_ars.timekeeper

import android.app.AlertDialog
import android.app.Dialog
import android.os.Bundle
import androidx.fragment.app.DialogFragment

class ConfirmationDialog : DialogFragment() {
    private lateinit var fConfirmAction: (() -> Unit)
    private var fMessage: Int = 0
    private var fPositiveMessage: Int = 0
    private var fNegativeMessage: Int = 0

    fun setOptions(confirmAction: () -> Unit, message: Int, positiveMessage: Int, negativeMessage: Int) {
        fConfirmAction = confirmAction
        fMessage = message
        fPositiveMessage = positiveMessage
        fNegativeMessage = negativeMessage
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val builder = AlertDialog.Builder(activity)
        builder.setMessage(fMessage)
                .setPositiveButton(fPositiveMessage) { _, _ -> fConfirmAction.invoke() }
                .setNegativeButton(fNegativeMessage, null)
        return builder.create()
    }
}